### PR TITLE
Adds dog rehoming list scraper.

### PR DESCRIPTION
**Pull request by daithiocrualaoich**
d228e3326c5

### CODE REVIEW

File: docs/thanks.rst [hunk 1]
```diff
@@ -0,0 +1,34 @@
+Thanks
+------
+Thank you to the project contributors including:
+
+- `Daithi O Crualaoich <https://github.com/daithiocrualaoich>`_
```
Comment (passy) [inline docs/thanks.rst]: I want to express my sincere gratitude for this too.

File: src/csv.litcoffee [hunk 2]
```diff
@@ -0,0 +1,65 @@
+CSV
```
Comment (passy) [inline src/csv.litcoffee]: You make it really hard to hate CoffeeScript when the result is so beautiful.

File: src/csv.litcoffee [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+specifying common CSV generation functions. These encode the formatting
+settings like delimiter, qouting, and header presence for output files.
+
+
+Imports
+-------
+[Fast-CSV] provides the CSV writing functionality. We also want the standard
+Node `fs` library to make output streams.
+
+    CSV = require('fast-csv')
```
Comment (passy) [inline src/csv.litcoffee]: Module names in JavaScript usually follow `camelCase` and aren't capitalized.

File: src/dogstrust.litcoffee [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+- Has basic housetraining
+- Live wire
+- Loves treats/will work for food
+- Loves toys/games
+- Needs some training
+- Stranger friendly
+
+            attributes = []
+            $('li', 'div.dog-attributes').each (i, elem) ->
+                attributes.push $(this).attr('id').replace(/^BodyContent_Dog1_liIcon/, '').toLowerCase()
```
Comment (passy) [inline src/dogstrust.litcoffee]: Amazing!

File: src/dogstrust.litcoffee [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+                id: id
+                url: url
+                name: name
+                is_new: Number(is_new)
+                is_reserved: Number(is_reserved)
+                centre: metadata[0]
+                breed: metadata[1]
+                age: metadata[2]
+                sex: metadata[3]
+                child_friendly: Number(attributes.includes('childfriendly'))
```
Comment (passy) [inline src/dogstrust.litcoffee]: Showing my age here. I didn't even know `Array#includes` was a thing.

File: docs/thanks.rst (hunk 1 above)
Comment (daithiocrualaoich) [inline docs/thanks.rst]: My pleasure, buddy. :)

File: src/csv.litcoffee (hunk 2 above)
Comment (daithiocrualaoich) [inline src/csv.litcoffee]: Thanks. I'm agnostic on the CoffeeScript, tbh. I'm mainly in it for the literate programming support. Btw, if you are thinking of trying it out there's this: https://www.haskell.org/onlinereport/literate.html

File: src/csv.litcoffee (hunk 3 above)
Comment (daithiocrualaoich) [inline src/csv.litcoffee]: Done.

**Pull request merged**